sda/processor.go

```diff
@@ -98,56 +108,47 @@ func (p *ServiceProcessor) Process(packet *network.Packet) {
 	log.Panic("Cannot handle message.")
 }
 
-// ProcessClientRequest takes a request from a client, calculates the reply
-// and sends it back.
-func (p *ServiceProcessor) ProcessClientRequest(si *network.ServerIdentity,
-	cr *ClientRequest) {
-	// unmarshal the inner message
-	mt, m, err := network.UnmarshalRegisteredType(cr.Data,
-		network.DefaultConstructors(network.Suite))
-	if err != nil {
-		log.Error("Err unmarshal client request:" + err.Error())
-		return
-	}
-	reply := p.GetReply(si, mt, m)
-	if err := p.SendRaw(si, reply); err != nil {
-		log.Error(err)
-	}
+func (i *ServiceProcessor) NewProtocol(tn *TreeNodeInstance, conf *GenericConfig) (ProtocolInstance, error) {
```

Comment:
To avoid
`// NewProtocol creates a protocol for stat, as you can see it is simultanously absolutely useless and regrettably necessary.`